### ISSUE TITLE
No inspection for while + array_shift to array_slice

### ISSUE THREAD

**Issue by vladyslavstartsev** [kalessil/phpinspectionsea#1025]
First, thank you for reporting a bug or making a request. That takes time and we appreciate that!

Let's start with common details:
 
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Feature request |
| Plugin         | Php Inspections (EA Extended) , 3.0.6 |
| Language level | PHP 7.2 |

Current behaviour (description/screenshot:)
--
no inspection for this kind of case

```php
$toDelete = count($foos) - self::MAX_FOO;
while ($toDelete > 0) {
    --$toDelete;
    array_shift($foos);
}
```

Expected behaviour
--
the code that is in `Current behaviour` should be converted to this

```php
$toDelete = count($foos) - self::MAX_FOO;
$foos = array_slice($foos, $toDelete);
```

Environment details
---
Information from Help -> About dialog (contains IDE and it's version details).

PhpStorm 2018.2.2
Build #PS-182.4129.45, built on August 22, 2018
JRE: 1.8.0_152-release-1248-b8 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.8.0-53-generic
